identity-service/src/routes/instagram.js

```diff
@@ -21,30 +21,70 @@ module.exports = function (app) {
         'client_id': config.get('instagramAPIKey'),
         'client_secret': config.get('instagramAPISecret'),
         'grant_type': 'authorization_code',
-        'redirect_uri': `${req.headers.origin}`,
+        'redirect_uri': config.get('instagramRedirectUrl'),
         code
       }
     }
 
     try {
       const res = await doRequest(reqObj)
-      const igUser = JSON.parse(res)
+      const authAccessToken = JSON.parse(res)
       let {
-        access_token: accessToken,
-        user: profile
-      } = igUser
-      if (!accessToken || !profile) return errorResponseBadRequest(new Error('invalid code'))
+        access_token: accessToken
+      } = authAccessToken
+
+      const instagramAPIUser = await doRequest({
+        method: 'get',
+        url: 'https://graph.instagram.com/me',
+        qs: {
+          'fields': 'id,username,media_count,account_type',
+          'access_token': accessToken
+        }
+      })
+      const igUser = JSON.parse(instagramAPIUser)
+      if (igUser.error) {
+        return errorResponseBadRequest(new Error(igUser.error.message))
+      }
+
+      const igMetadataResponse = await doRequest({
+        method: 'get',
+        url: `https://www.instagram.com/${igUser.username}/?__a=1`
```

Comment:
let's comment this with a stack overflow link or something so we can future proof it